sycl/include/sycl/detail/string_view.hpp

```diff
@@ -0,0 +1,67 @@
+//==-------------- string_view.hpp - SYCL standard header file -------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <string>
+
+#pragma once
+
+namespace sycl {
+inline namespace _V1 {
+namespace detail {
+
+// This class and detail::string class are intended to support
+// different ABIs between libsycl and the user program.
+// This class is not inteded to replace std::string_view for general purpose
+// usage.
+class string_view {
+  const char *str = nullptr;
+
+public:
+  string_view() noexcept = default;
+  string_view(const string_view &strn) noexcept = default;
+  string_view(std::string_view strn) noexcept : str(strn.data()) {}
+  string_view(string_view &&strn) noexcept : str(strn.data()) {}
+
+  string_view &operator=(string_view &&strn) noexcept {
```

Comment:
Likely so are (move-)assignment operators, possibly even just by omission.

Reply:
We have to keep the move assignment operator.
It was requested by Victor.

Reply:
In `string_view`, it can be defaulted as well as other special member functions as no resource management is happening here